### PR TITLE
Ccd 4197 

### PR DESCRIPTION
**Pull request by padmapriyanalam**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CCD-4197

### Change description ###
Update implementation of CategoriesAndDocuments to Move Document to 'UnCategorised' category
CCD-4197 is based on CCD-4196 which has been addressed in the [PR](https://github.com/hmcts/ccd-data-store-api/pull/2232)

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ccd/domain/service/casefileview/CategoriesAndDocumentsService.java [hunk 1]
```diff
@@ -175,9 +179,16 @@ private Document buildDocument(final String attributePath, @NonNull final Map<St
...
+                                   final String categoryOnFieldDefinition,
+                                   final List<CategoryDefinition> categories) {
+        if (categoryOnDocument != null && !categories.stream()
+            .anyMatch(category ->
+                category.getCategoryId().equals(categoryOnDocument))) {
+            return UNCATEGORISED_KEY;
+        } else {
+            return Optional.ofNullable(categoryOnDocument)
+                .orElseGet(() -> resolveDocumentCategory(categoryOnFieldDefinition));
+        }
```
Comment (aktaskaan) [inline src/main/java/uk/gov/hmcts/ccd/domain/service/casefileview/CategoriesAndDocumentsService.java:191]:
```suggestion
        boolean isDocumentCategoryPresent = categoryOnDocument != null &&
            categories.stream()
                .anyMatch(category -> category.getCategoryId().equals(categoryOnDocument));

        return isDocumentCategoryPresent ? categoryOnDocument : resolveDocumentCategory(categoryOnFieldDefinition);
```
Please check this out. I think these makes the code more maintainable and easier to understand.

File: src/main/java/uk/gov/hmcts/ccd/domain/types/DocumentValidator.java [hunk 2]
```diff
@@ -151,19 +150,6 @@ private List<ValidationResult> validateCategoryId(final String dataFieldId,
...
-        String categoryIdValue = categoryId.textValue();
-        final boolean caseTypeContainsKnownCategoryId = categoryList.stream()
-            .anyMatch(category ->
-                category.getCategoryId().equals(categoryIdValue));
-
-        if (!caseTypeContainsKnownCategoryId) {
-            LOG.error("{} value not recognised as a valid Case Category", CATEGORY_ID);
-            return Collections.singletonList(new ValidationResult(
-                CATEGORY_ID + " value not found", dataFieldId));
-        }
```
Comment (aktaskaan) [inline src/main/java/uk/gov/hmcts/ccd/domain/types/DocumentValidator.java:166]: Could you please provide additional details about the reason for deleting this?

File: src/test/java/uk/gov/hmcts/ccd/domain/service/casefileview/CategoriesAndDocumentsServiceTest.java [hunk 3]
```diff
@@ -111,7 +112,10 @@ void testShouldTransformAll() {
     void testShouldResolveDocumentCategory(final String categoryOnDocument,
```
Comment (aktaskaan) [inline src/test/java/uk/gov/hmcts/ccd/domain/service/casefileview/CategoriesAndDocumentsServiceTest.java:112]: Please add more unit test to cover all possible conditions for the results of **underTest.resolveDocumentCategory**

**Review by aktaskaan — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/ccd/domain/types/DocumentValidator.java (hunk 2 above)
Comment (padmapriyanalam) [reply src/main/java/uk/gov/hmcts/ccd/domain/types/DocumentValidator.java:166]: This has been branched out of CCD-4196 which has been addressed in the [PR](https://github.com/hmcts/ccd-data-store-api/pull/2232)

**Review by JamesCollettCGI — APPROVED** [review]

**Pull request merged**